File: Modul 4/soal_4/PRAK404-2410817310003-Muhammad_Guntur_Ricky_Adhitya.py
def operasi(pilihan):
    nilai_pertama = float(input("Masukkan Nilai Pertama : "))
    nilai_kedua = float(input("Masukkan Nilai Kedua : "))
    hasil = float()
    operasi_t = "penjumlahan", "pengurangan", "perkalian", "pembagian"

    match pilihan:
            case 1 :
                hasil = nilai_pertama + nilai_kedua
            case 2 :
                hasil = nilai_pertama - nilai_kedua
            case 3 :
                hasil = nilai_pertama * nilai_kedua
            case 4 : 
                hasil = nilai_pertama / nilai_kedua
    
    print("Hasil %s antara %.2f dengan %.2f adalah %.2f"%(operasi_t[pilihan-1], nilai_pertama, nilai_kedua, hasil))


while True:
    print("Pilih Program\n1.Penjumlahan\n2.Pengurangan\n3.Perkalian\n4.Pembagian\n5.Exit")
    pilihan = int(input("Masukkan Pilihan : "))

    match pilihan:
        case 1 :
            operasi(pilihan)
        case 2 :
            operasi(pilihan)
        case 3 :
            operasi(pilihan)
        case 4 :
            operasi(pilihan)
        case 5 :
            print("Terimakasih, telah menggunakan kalkulator Guntur")
            break
        case _: 
            print("Input anda salah, silahkan coba lagi")

File: Modul 4/soal_4/PRAK404-2410817310003-Muhammad_Guntur_Ricky_Adhitya.c
#include <stdio.h>

void operasi(int pilihan){
    float nilai_pertama;
    printf("Masukkan nilai pertama : ");
    scanf("%f", &nilai_pertama);

    float nilai_kedua;
    printf("Masukkan nilai kedua : ");
    scanf("%f", &nilai_kedua);
    float hasil = 0;
    char operasi[][32] = {"penjumlahan", "pengurangan", "perkalian", "pembagian"};
    switch (pilihan)
    {
    case 1:
        hasil = nilai_pertama + nilai_kedua;
        break;
    
    case 2:
        hasil = nilai_pertama - nilai_kedua;
        break;

    case 3:
        hasil = nilai_pertama * nilai_kedua;
        break;

    case 4:
        hasil = nilai_pertama / nilai_kedua;
        break;

    default:
        break;
    }

    printf("Hasil %s antara %.2f dengan %.2f adalah %.2f\n", operasi[pilihan-1], nilai_pertama, nilai_kedua, hasil);
}

int main(){

    while(1){
        printf("Pilih Program\n1.Penjumlahan\n2.Pengurangan\n3.Perkalian\n4.Pembagian\n5.Exit\n");

        int pilihan;
        printf("Masukkan Pilihan : ");
        scanf("%d", &pilihan);
        
        if(pilihan == 5){
            printf("Terimakasih, telah menggunakan kalkulator Guntur\n");
            break;
        }else if(pilihan > 0 && pilihan < 5){
            operasi(pilihan);
        }else{
            printf("Input anda salah, silahkan coba lagi\n");
        }
    }
    return 0;
}